sdk/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.dataflow.sdk.transforms.display;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Static display metadata associated with a {@link PTransform}. Display data is useful for
+ * pipeline runner UIs and diagnostic dashboards to display details about {@link PTransform}s
+ * that make up a pipeline.
+ *
+ * <p>{@link PTransform}s specify their display data by implementing the {@link HasDisplayData}
+ * interface.
+ */
+public class DisplayData {
+  private static final DisplayData EMPTY = new DisplayData(Maps.<Identifier, Item<?>>newHashMap());
+  private final ImmutableMap<Identifier, Item<?>> entries;
+
+  private DisplayData(Map<Identifier, Item<?>> entries) {
+    this.entries = ImmutableMap.copyOf(entries);
+  }
+
+  /**
+   * Default empty {@link DisplayData} instance.
+   */
+  public static DisplayData none() {
+    return EMPTY;
+  }
+
+  /**
+   * Collect the {@link DisplayData} for a {@link PTransform}. This will traverse all subcomponents
+   * specified via {@link Builder#include} in the given transform.
+   */
+  public static DisplayData from(PTransform<?, ?> transform) {
+    checkNotNull(transform);
+    return InternalBuilder.forRoot(transform).build();
+  }
+
+  public Collection<Item<?>> items() {
```

Comment:
What exactly DisplayData means when interpreted as a map or "items" isn't obvious. While we're at it, it may be worth calling it `getItems`, depending on the use.